packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts

```diff
@@ -0,0 +1,106 @@
+/**
+ * Internal dependencies
+ */
+
+/**
+ * Type dependencies
+ */
+import { ElementHandle, Frame, Page } from 'playwright';
+import { BaseContainer } from '../base-container';
+
+/**
+ * Represents an instance of the WPCOM's Gutenberg editor page.
+ *
+ * @augments {BaseContainer}
+ */
+export class GutenbergEditorPage extends BaseContainer {
+	frame!: Frame;
+
+	// Page and Frame related selectors.
+	editorSelector = 'div.main.main-column.calypsoify.is-iframe > iframe';
+
+	// Editor element selectors.
+	titleSelector = `.editor-post-title__input`;
+	appenderSelector = '.block-editor-default-block-appender';
+	paragraphSelector = 'p.block-editor-rich-text__editable:first-of-type';
+
+	// Publish flow selectors.
+	firstPublishButtonSelector = '.editor-post-publish-panel__toggle';
+	secondPublishButtonSelector =
```

Comment:
I'd keep those two above as `prePublishButtonSelector` and `publishButtonSelector`. Not a strong opinion here, just think that they describe the flow a bit better.

Reply:
I struggled with the naming of this a bit, and do not like neither the existing names nor my proposed names. There should be a better way to call them, perhaps by the location of where they are?

Perhaps `headerPublishButtonSelector` and `panelPublishButtonSelector`?

Reply:
I've renamed the selectors. Please let me know if you agree with the naming scheme - if not, I am happy to adopt your scheme.

Reply:
Naming is always the hardest part, I also struggle with it every time! 😄 
Maybe, if possible we should derive the name from the actual CSS selector, and if not then focus on the element function / type? Here's my take on these (notice the drop of `Selector` suffix as they're already under the `selectors` object):

```js
const selectors = {
	editorFrame: 'div.main.main-column.calypsoify.is-iframe > iframe',
	blockAppender: '.block-editor-default-block-appender',
	titleInput: '.editor-post-title__input',
	paragraphInput: 'p.block-editor-rich-text__editable:first-of-type',
	publishPanel: '.editor-post-publish-panel',
	publishPanelToggle: '.editor-post-publish-panel__toggle',
	publishButton: '.editor-post-publish-panel__header-publish-button button.editor-post-publish-button',
	viewPostButton: 'text=View Post',
};
```

Anyway, I'm not really strong on these but I would at least drop the `Selector` suffix 🗑️

Reply:
> I would at least drop the Selector suffix 🗑️

Good catch, it made sense when the selectors were in the class but now it's just repeating itself for no reason. Will push a revised commit today PDT.

With regard to names, let's codify that as the style guide going forward. I have plans to update the documentation with new Playwright style guide in #52464. I will push up a change to include name changes today.

Reply:
Implemented changes to selector names for the components in this PR. 

`LoginPage` still needs to migrate to the new paradigm.